### DISCUSSION TITLE
Wiring

### DISCUSSION THREAD

**Discussion by Festivejelly** [kachurovskiy/nanoels#137] [Q&A]
Looking at the readme for the H4 I came across this:

_Note: depending on your particular encoder, stepper driver and signal wire, H4 might not be able to provide sufficient voltage/current on the 5V output lines above due to the presence of a fuse and current-limiting resistors on the PCB. If your encoder isn't functioning or steppers are working erratically, consider supplying 5V from an external power source to the 5V lines above._

So are you saying I should provide the 5V to the drivers and the encoder from an external power supply, not from the 5v terminals on the H4 board? So do I leave the 5V terminals unconnected on the H4? Presumably I still need to ground the external 5V to the control board though?

**Comment by kachurovskiy** [comment]
This is optional - I didn't realize this might be needed until the last week when I came across a driver that would work very unreliably. Measuring the voltages on the driver inputs I figured that signal voltage is sagging due to too much resistance on the 5V line.

My encoder is happy with the 5V supplied from H4 but some encoders aren't - https://github.com/kachurovskiy/nanoels/discussions/136

> So do I leave the 5V terminals unconnected on the H4?

This is up to you, as long as it's the same 5V that powers H4 via the USB port it doesn't matter. You can just supply the 5V directly to the driver and encoder - without using the 5V ports on the back of the H4.

> Presumably I still need to ground the external 5V to the control board though?

It should be the same 5V that's coming in via the USB port and H4 would already have the ground from the USB.

Sorry for this needless complexity, I over-engineered the PCB here, will likely remove these current-limiting resistors and encoder fuse in the next batch.

**Reply by Festivejelly** [reply]
Oh I see. But the USB goes direct into the Nano so id need to pick it up somewhere off the board.

That said I've not tested my encoder yet to see if it even needs it. Im using an optical 600 type. Im just wiring it all up now my power supply has arrived. I guess I can do a test just by powering the Nano and the Encoder from my 5v USB.

**Reply by kachurovskiy** [reply]
Got it. Please let me know if there would be any problems.

**Reply by Festivejelly** [reply]
So this is the encoder I have. Ive tried plugging it in as per instructions and I dont seem to be getting any reading from it when I press the square. The angle doesnt seem to be changing https://www.amazon.co.uk/gp/product/B07R6GS3YY/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1

So should I use a jumper wire from the 5v from the USB to the 5v on the encoder?

**Reply by kachurovskiy** [reply]
Yes, please try powering encoder from the same power supply as H4. I would not recommend soldering something on the PCB though as it's very tight there already.

**Reply by Festivejelly** [reply]
Sorry for the basic questions, electronics is not my strong point. So I could splice my USB-C cable that goes into the Nano controller so that I have 5v coming off it like this:

![image](https://github.com/kachurovskiy/nanoels/assets/2408274/02f20a7a-4c1b-4aa2-8976-59c760df03ee)


and this 5v that ive spliced goes to the rotary encoder 5v right?

**Reply by kachurovskiy** [reply]
I guess that is one of the options though it's not ideal to splice the USB cable like that.

Usually a 5V power source is used that allows connecting more than one 5V output: https://www.aliexpress.com/item/1005002843829663.html

Alternatively you could get a split USB cable: https://www.aliexpress.com/item/1005005256074407.html

**Reply by Festivejelly** [reply]
The splitter would probably work the best because I intend to link the Nano to the raspberry pi.

**Reply by Festivejelly** [reply]
My janky splice on the USB cable seemed to do the trick. I'll do a more permanent solution soon, but for now this will do.

**Comment by vegetate7** [comment]
> Sorry for this needless complexity, I over-engineered the PCB here, will likely remove these current-limiting resistors and encoder fuse in the next batch.

If you going to change PCB, may be worth to add GND terminals to AUX1/AUX2? And. may be dedicatet POWER-IN terminals.
BTW, can I power mainboard from ENCODER's terminal 5v/GND? Just do not like an USB connector.

**Reply by kachurovskiy** [reply]
Changes to the PCB in the next batches are going to be very minor to address the problems covered in https://github.com/kachurovskiy/nanoels/blob/main/h4/parts.md

Significant changes would make it a different model, requiring new case, new round of certifications.

**Reply by jordanpayn1989** [reply]
Just incase anyone else has the same question of backfeeding power to the H4 via encoder inputs. I have had mine powered through them for about a month now using the same 5.5V fed to encoder from buck converter stepping down 42V. No issues so far🤷‍♂️
